subsys/bluetooth/audio/bap_broadcast_sink.c

```diff
@@ -430,6 +430,11 @@ static void broadcast_sink_add_src(struct bt_bap_broadcast_sink *sink)
 	if (err < 0) {
 		LOG_WRN("Failed to add sync as Receive State for sink %p: %d",
 			sink, err);
+		// If source was already added, update the valid SRC ID flag
+		if (err == -EALREADY ){
+			LOG_WRN("Updating sink->flags");
+			atomic_set_bit(sink->flags,BT_BAP_BROADCAST_SINK_FLAG_SRC_ID_VALID);
+		}
```

Comment:
Make sure to remove this when https://github.com/zephyrproject-rtos/zephyr/pull/63982 is merged. Alternative you can also apply the fix from that PR here